### ISSUE TITLE
[appletrailers] search result triggers HTTP 404 Error

### ISSUE THREAD

**Issue by farmerchris** [rg3/youtube-dl#7047]
Executing a command returning a particular search result throws an exception on Mac OS X, newest version of youtube-dl:

```$  youtube-dl --verbose --get-filename --default-search gvsearch2  "large apple"```

```
[debug] System config: []
[debug] User config: [u'--format', u'mp4']
[debug] Command-line args: [u'--verbose', u'--get-filename', u'--default-search', u'gvsearch2', u'large apple']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.09.28
[debug] Python version 2.7.10 - Darwin-14.5.0-x86_64-i386-64bit
[debug] exe versions: avconv 11.4, avprobe 11.4, ffmpeg 2.8, ffprobe 2.8, rtmpdump 2.4
[debug] Proxy map: {}
Pruning a large old apple tree-Qar8TmsobPk.mp4
ERROR: Unable to download XML: HTTP Error 404: Not Found (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 327, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1865, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
```

In this case it is the second search result from "large apple" that throws the error. Any other search term, or searching for just 1 item shows no error. I can't see what that search result is, so I can only hope your search returns the same results as mine.

**Comment by farmerchris** [comment]
A little more data when omitting the `--get-filename` option. Seems as if the offending link is the Apple Trailers page for "Over the Hedge".

```
[debug] System config: []
[debug] User config: [u'--format', u'mp4']
[debug] Command-line args: [u'--verbose', u'--default-search', u'gvsearch2', u'large apple']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.09.28
[debug] Python version 2.7.10 - Darwin-14.5.0-x86_64-i386-64bit
[debug] exe versions: avconv 11.4, avprobe 11.4, ffmpeg 2.8, ffprobe 2.8, rtmpdump 2.4
[debug] Proxy map: {}
[video.google:search] gvsearch:large apple: Downloading result page 1
[download] Downloading playlist: large apple
[video.google:search] playlist large apple: Collected 2 video ids (downloading 2 of them)
[download] Downloading video 1 of 2
[youtube] Qar8TmsobPk: Downloading webpage
[youtube] Qar8TmsobPk: Downloading video info webpage
[youtube] Qar8TmsobPk: Extracting video information
[youtube] Qar8TmsobPk: Downloading DASH manifest
[youtube] Qar8TmsobPk: Downloading DASH manifest
[debug] Invoking downloader on u'https://r8---sn-ufuxaxjvh-q4fe.googlevideo.com/videoplayback?id=o-AMz4jcNgHxsDLwKJ-d5sQrvouWQoX5JFKpy4P1K5r_i9&pl=21&mm=31&mn=sn-ufuxaxjvh-q4fe&ms=au&signature=69B2BD3FE1ED91A0DCD323A4A004FBA89E1A6728.843986E9B9862435A3B5222E81AD8D558128978B&mt=1443820766&sparams=dur%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cpcm2%2Cpcm2cms%2Cpl%2Cratebypass%2Crequiressl%2Csource%2Cupn%2Cexpire&mv=m&ip=168.215.167.148&pcm2=yes&requiressl=yes&ratebypass=yes&expire=1443842452&sver=3&ipbits=0&dur=469.762&itag=22&mime=video%2Fmp4&lmt=1389259956459837&key=yt6&fexp=9406513%2C9407157%2C9408491%2C9408710%2C9409069%2C9412773%2C9412846%2C9414764%2C9415365%2C9415485%2C9415521%2C9416023%2C9416126%2C9416729%2C9417707%2C9418153%2C9418392%2C9418448%2C9418494%2C9419321%2C9419446%2C9419786%2C9419862%2C9420348%2C9420981%2C9421013%2C9421369%2C9421410%2C9421655%2C9421943&pcm2cms=yes&upn=JGItI3NQtD0&source=youtube&initcwndbps=2691250'
[download] Pruning a large old apple tree-Qar8TmsobPk.mp4 has already been downloaded
[download] 100% of 166.18MiB
[download] Downloading video 2 of 2
[AppleTrailers] overthehedge: Downloading XML
ERROR: Unable to download XML: HTTP Error 404: Not Found (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 327, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1865, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
```

**Comment by yan12125** [comment]
Should be fixed since commit 494172d2e5b2d5b6f309b42e1a2bd7108aed40de. Feel free to open new issues if similar problems occur again.
